### ISSUE TITLE
Warning when fitting without errors is misleading

### ISSUE THREAD

**Issue by cverstege** [PhiLFitters/kafe2#209]
Especially this line:
https://github.com/PhiLFitters/kafe2/blob/742a3f577bf1a0cd70101b2dd3abda46267de053/kafe2/fit/_base/cost.py#L330

I know, when fitting without uncertainties, one should choose `chi2_no_errors` or similar as cost function.
For inexperienced users, this warning is misleading, as a fit still fails when not setting any errors and converges when manually setting the errors to one.
```python
from kafe2 import XYFit

x = [0,1,2,3,4]
y = [0,1,2,3,4]

fit1=XYFit([x, y]) #, cost_function="chi2_no_errors")
fit1.do_fit()
fit2=XYFit([x, y])
fit2.add_error('y', 1.0)
fit2.do_fit()
print(fit1.get_result_dict()['cost'])
print(fit2.get_result_dict()['cost'])
```

**Comment by cverstege** [comment]
I guess this is fixed by 157d955fb68e38026409ffe0779977f05e93ca00. Please confirm @JohannesGaessler

**Comment by JohannesGaessler** [comment]
Yes, this should be fixed; I just forgot to close the issue.
